src/conduit.js

```diff
@@ -0,0 +1,25 @@
+class Conduit {
+  constructor(inlet, outlet) {
+    this.inlet = inlet;
+    this.outlet = outlet;
+  }
+
+  disconnect = () => {
+    this.inlet.onDisconnect(this);
+    this.outlet.onDisconnect(this);
+  };
+
+  connect = () => {
+    this.inlet.onConnect(this);
+    this.outlet.onConnect(this);
+  };
+
+  getId = () => `${this.inlet.getId()}-${this.outlet.getId()}`;
+  getInlet = () => this.inlet;
+  getOutlet = () => this.outlet;
+  update = () => this.outlet.forceRender();
+  simplify = () => ({ inlet: this.inlet, outlet: this.outlet });
```

Comment:
IMHO simplify is a very obscure name ^_^U. Here and in all the places where it is used.